lib/cdo/contact_rollups/v2/pardot.rb

```diff
@@ -176,18 +180,32 @@ def batch_update_remaining_prospects
   # @param [String] api_endpoint
   # @param [Array<Hash>] prospects
   # @param [Boolean] eager_submit if is true, triggers submitting request immediately
+  # @param [String] batch is to create new prospects ('create') or update existing ones ('update')
   # @return [Array<Array>] two arrays, one for all submitted prospects and one for Pardot errors
-  def process_batch(api_endpoint, prospects, eager_submit)
+  def process_batch(api_endpoint, prospects, eager_submit, batch_type)
     return [], [] unless prospects.present?
 
     submissions = []
     errors = []
     url = self.class.build_batch_url api_endpoint, prospects
 
     if url.length > URL_LENGTH_THRESHOLD || prospects.size == MAX_PROSPECT_BATCH_SIZE || eager_submit
-      # TODO: Rescue Net::ReadTimeout from submit_prospect_batch and tolerate a certain number of failures.
-      #   Use an instance variable to remember the number of failures.
-      errors = self.class.submit_batch_request api_endpoint, prospects
+      if @dry_run
+        unless @dry_run_batches_built.include? batch_type
+          puts "[Sample Batch] Prospects to sync to Pardot: #{prospects.length}"
+          puts "[Sample Batch] Query string:\n#{url}"
+          puts '[Sample Batch] Contacts to be added:'
```

Comment:
Because data is in prospect format (with db_* fields) and could be either _added_ or _updated_ to Pardot
```suggestion
          puts '[Sample Batch] Prospects to be synced:'
```